src/tests.rs

```diff
@@ -149,7 +149,7 @@ fn named_from_files_pass_tests() {
         let start_time = std::time::Instant::now();
         let crates = crates_data_from_file(&case);
         let mut dp = Index::new(&crates);
-        let root = new_bucket(&name, (&ver).into(), true);
+        let root = new_bucket("test-root", (&ver).into(), true);
         if !check(&mut dp, root, &ver) {
             dp.make_index_ron_file();
```

Comment:
`make_index_ron_file` writes down the portion of the index that was actually accessed. This creates a smaller test file that can be recursively minimized to identify the root cause the problem. It is also used in the other tests in this module and in the `process_carte_version`. It will need to be updated to write a file that these tests actually work with. When the smaller problem comes from a test case and still involves the root crate, this does not cause a problem. But sometimes some crate in the middle of the tree is sufficient to reproduce the problem (like `all_vers_in_files_pass_tests` checks for) or the underlying problem comes from crates.io where the root crate is not called "test-root". The ability to automatically go from "something on crates.io failed" to "I have a test case that reproduces the issue" has been very important.

- We could change `make_index_ron_file` to always rename the root crate as "test-root". If we have additional bugs that involve circular dependencies, then we will also need to rename the root package in the dependencies of other packages in the index.
- We could probe for the name from the file name (in case it's an automatically generated case) and if the file does not contain that package fallback to "test-root" (in case it's manually curated case). This will involve looking at the contents the file to determine what the root crate is, which wouldn't be too painful with the right abstraction.
- We could make a more complicated grammar in the filename. So if it matches `name@version.ron` it uses the existing steam otherwise it assumes that the name is "test-root" and ver = 0.0.1. Or perhaps it is `name+comment@version.ron` where `comment` is optional and ignored.

I'm not sure which option is best. What do you think?

Reply:
I think we can keep the old behavior for now.